modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/config/CliConfig.scala

```diff
@@ -68,72 +64,5 @@ object CliConfig {
    *         correctly
    */
   def parse(argv: Seq[String]): ValidatedNel[String, CliConfig] =
-    parser.parse(argv).leftMap(_.show).toValidatedNel.andThen(transform)
-
-
-  /**
-   * Initial raw configuration parsed from CLI arguments
-   * Could be invalid, supposed to be validated and transformed
-   * into `CliConfig`
-   *
-   * @param config base64-encoded Snowplow config.yml
-   * @param resolver base64-encoded Iglu Resolver JSON
-   * @param dryRun if RDB Loader should just discover data and print SQL
-   */
-  private[config] case class RawConfig(config: String,
-                                       resolver: String,
-                                       dryRun: Boolean)
-
-  type Parsed[A] = ValidatedNel[String, A]
-
-  /** Wrapper for any Base64-encoded entity */
-  case class Base64Encoded[A](decode: A)
-
-  implicit def base64EncodedInstance[A: Argument]: Argument[Base64Encoded[A]] = new Argument[Base64Encoded[A]] {
-    def read(string: String): ValidatedNel[String, Base64Encoded[A]] = {
-      val str = Validated
-        .catchOnly[IllegalArgumentException](new String(Base64.getDecoder.decode(string), StandardCharsets.UTF_8))
-        .leftMap(_.getMessage)
-        .toValidatedNel
-      str.andThen(Argument[A].read).map(a => Base64Encoded(a))
-    }
-
-    def defaultMetavar: String = "base64"
-  }
-
-  /**
-   * Validated and transform initial raw cli arguments into
-   * ready-to-use `CliConfig`, aggregating errors if any
-   *
-   * @param rawConfig initial raw arguments
-   * @return application config in case of success or
-   *         non empty list of config errors in case of failure
-   */
-  private[config] def transform(rawConfig: RawConfig): ValidatedNel[String, CliConfig] = {
-    val config: Parsed[Config[StorageTarget]] = base64decode(rawConfig.config).flatMap(Config.fromString).toValidatedNel
-    val client: Parsed[(Json, Client[Id, Json])] = loadResolver(rawConfig.resolver).toValidatedNel
-
-    (config, client).mapN {
-      case (c, (j, _)) => CliConfig(c, rawConfig.dryRun, j)
-    }
-  }
-
-  /**
-   * Safely decode base64 string into plain-text string
-   *
-   * @param string string, supposed to be base64-encoded
-   * @return either error with full description or
-   *         plain string in case of success
-   */
-  private def base64decode(string: String): Either[String, String] =
-    Either
-      .catchOnly[IllegalArgumentException](new String(Base64.getDecoder.decode(string), StandardCharsets.UTF_8))
-      .leftMap(_.getMessage)
-
-  /** Decode Iglu Resolver and associated JSON config */
-  private def loadResolver(resolverConfigB64: String): Either[String, (Json, Client[Id, Json])] = {
```

Comment:
The `loadResolver` method that you deleted contained a `Client.parseDefault`, but the client was never used.

I wonder if that was deliberate though?! By parsing the resolver at this early stage it means the app fails at this early stage if the resolver is unexpected.

Honestly, I don't know if the original design decision was deliberate or not.  In any case, the app still fails very quickly if the resolver is invalid, so I don't mind removing it here.

Reply:
Pretty sure it's deliberate. We had a similar check in BQ Loader: https://github.com/snowplow-incubator/snowplow-bigquery-loader/commit/6595d5b60fa0e2d31c6714909ca506b6701e86fb#diff-4134a56e1b2220f83954dc00124ebf3abcdebc5caf8fa835f5bfae75492acc16L115

I changed it slightly to not use a `Client`: https://github.com/snowplow-incubator/snowplow-bigquery-loader/blob/38d573e93a4470cbde2e3f3ecf0fad83041f3f94/modules/common/src/main/scala/com/snowplowanalytics/snowplow/storage/bigquery/common/config/CliConfig.scala#L68